pkg/policy/api/rule.go

```diff
@@ -70,6 +83,63 @@ func NewRule() *Rule {
 	return &Rule{}
 }
 
+// MarshalJSON returns the JSON encoding of Rule r. We need to overwrite it to
+// enforce omitempty on the EndpointSelector nested structures.
```

Comment:
Current approach requires a bit more code, but changing the type to avoid a serialization issue felt somewhat like a hack. Plus, being quite new to Golang, I wasn't too sure I understood the consequences.
Was there a reason for not using a pointer at first? Or does that really not matter and I should just switch it?